blogs/2021/01/ext-bisect.md

```diff
@@ -0,0 +1,40 @@
+# “Just like git-bisect, but for VS Code extensions.”
+
+The true power of VS Code is its extensions: theme extensions add colors and icons, language extensions provide IntelliSense and enable code navigation, debugger extensions enable you to drill into bugs. There are extensions that play music, some that show stock tickers, and there are extensions that enable collaborative work across locations and time zones. The VS Code marketplace hosts more than 23K extensions and it is not uncommon that users have 50+ extensions installed. With so many extensions bugs are inevitable and instead of denial we want to make trouble shooting easy.
+
+We love extensions. There are no bad extensions. Like all software, extensions have bugs and feature gaps and for reading ease and artificial drama, let’s use “bad extensions”. Understand it is an extension showing unwanted behavior. In the fact, we have designed VS Code with bad extensions in mind and therefore run them in a separate [process]( https://code.visualstudio.com/api/advanced-topics/extension-host). The guarantee is that the cursor always blinks and that you can always save your work. However, not all bad extensions crash or freeze a process, sometimes it is just a small, sporadically showing, annoyance.
+
+Because there are no bad extensions, we have created one: [Extension Bisect Demo](https://marketplace.visualstudio.com/items?itemName=jrieken.bisectdemo). When installed, it will annoyingly reset your cursor whenever you reach the word “bisect”. Use that extension to follow along.
+
+Finding a bad extension is easy: Open the extension view, [disable an extension]( https://code.visualstudio.com/docs/editor/extension-gallery#_disable-an-extension), reload, and check for the issue. If the problem is gone that extension is bad and you are done. Otherwise, re-enable the extension and repeat the process with the next extension.
+
+![Disabling an extension step by step](./disable_manually.png)
+
+If you are lucky the first extension is bad, if you are unlucky it’s the last extension. Using computer science language this means with `N` extensions you have a worst-case of `O(N)`, and an average-case of `O(N/2)`. Because this algorithm is operated by a human even small values of `N` are laborious and mindless. This is where extension bisect comes into play. It is much better in the worst- and average-case because it disables extensions by halves. This is like the problem of finding out which commit in git has caused a regression. Git has the [`git bisect`](https://git-scm.com/docs/git-bisect)-command for that and that inspired us to do the same for extensions.
+
+Let’s use a sample:  I have 24 extensions installed (this command tells you: `code --list-extensions | wc -l`) and the 8th extension is bad. We know that the iterative approach requires 8 steps. What about bisect?
```

Comment:
Should we also have a Windows version `code --list-extensions | find /v /c ""`?
`wc` only works on bash

Reply:
Yeah, I will remove that as it isn't critical and since there multiple ways to figure out many extensions you have.